qiskit/transpiler/passes/synthesis/solovay_kitaev_utils.py

```diff
@@ -0,0 +1,634 @@
+    z = 1 / np.sqrt(np.linalg.det(u2_matrix))
+    su2_matrix = z * u2_matrix
+    phase = np.arctan2(np.imag(z), np.real(z))
+
+    return su2_matrix, phase
+
+
+def compute_euler_angles_from_so3(matrix: np.ndarray) -> Tuple[float, float, float]:
+    """Computes the Euler angles from the SO(3)-matrix u.
+
+    Uses the algorithm from Gregory Slabaugh,
+    see `here <https://www.gregslabaugh.net/publications/euler.pdf>`_.
+
+    Args:
+        matrix: The SO(3)-matrix for which the Euler angles need to be computed.
+
+    Returns:
+        Tuple phi, theta, psi\n
+        where phi is rotation about z-axis, theta rotation about y-axis\n
+        and psi rotation about x-axis.
+    """
+    matrix = np.round(matrix, decimals=7)
+    if (matrix[2][0] != 1 and matrix[2][1] != -1):
+        theta = -math.asin(matrix[2][0])
+        psi = math.atan2(matrix[2][1] / math.cos(theta),
+                         matrix[2][2] / math.cos(theta))
+        phi = math.atan2(matrix[1][0] / math.cos(theta),
+                         matrix[0][0] / math.cos(theta))
+        return phi, theta, psi
+    else:
+        phi = 0
+        if matrix[2][0] == 1:
+            theta = math.pi/2
+            psi = phi + math.atan2(matrix[0][1], matrix[0][2])
+        else:
+            theta = -math.pi/2
+            psi = -phi + math.atan2(-matrix[0][1], -matrix[0][2])
+        return phi, theta, psi
+
+
+def compute_su2_from_euler_angles(angles: Tuple[float, float, float]) -> np.ndarray:
+    """Computes SU(2)-matrix from Euler angles.
+
+    Args:
+        angles: The tuple containing the Euler angles for which the corresponding SU(2)-matrix
+            needs to be computed.
+
+    Returns:
+        The SU(2)-matrix corresponding to the Euler angles in angles.
+    """
+    phi, theta, psi = angles
+    uz_phi = np.array([[np.exp(-(1/2)*phi*1j), 0],
+                       [0, np.exp((1/2)*phi*1j)]], dtype=complex)
+    uy_theta = np.array([[math.cos(theta/2), math.sin(theta/2)],
+                         [-math.sin(theta/2), math.cos(theta/2)]], dtype=complex)
+    ux_psi = np.array([[math.cos(psi/2), math.sin(psi/2)*1j],
+                       [math.sin(psi/2)*1j, math.cos(psi/2)]], dtype=complex)
+    return np.dot(uz_phi, np.dot(uy_theta, ux_psi))
+
+
+def convert_su2_to_so3(matrix: np.ndarray) -> np.ndarray:
+    """Computes SO(3)-matrix from input SU(2)-matrix.
+
+    Args:
+        matrix: The SU(2)-matrix for which a corresponding SO(3)-matrix needs to be computed.
+
+    Returns:
+        The SO(3)-matrix corresponding to ``matrix``.
+
+    Raises:
+        ValueError: if ``matrix`` is not an SU(2)-matrix.
+    """
+    if matrix.shape != (2, 2):
+        raise ValueError(
+            'Conversion from SU2 called on matrix of shape', matrix.shape)
+
+    if abs(np.linalg.det(matrix) - 1) > 1e-4:
+        raise ValueError(
+            'Conversion from SU2 called on determinant of', np.linalg.det(matrix))
+
+    matrix = matrix.astype(np.complex)
```

Comment:
Aligned to https://github.com/Qiskit/qiskit-terra/pull/5788
```suggestion
    matrix = matrix.astype(complex)
```